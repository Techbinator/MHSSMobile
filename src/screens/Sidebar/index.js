import React, { Component } from 'react';
import { TouchableOpacity } from 'react-native';
import { StackActions, NavigationActions } from 'react-navigation';
import {
  Container,
  Content,
  Text,
  Icon,
  ListItem,
  Thumbnail,
  View,
} from 'native-base';
import { Grid, Col } from 'react-native-easy-grid';

import styles from './style';
const resetAction = StackActions.reset({
  index: 0,
  actions: [NavigationActions.navigate({ routeName: 'Login' })],
});

class SideBar extends Component {
  render() {
    const navigation = this.props.navigation;
    return (
      <Container>
        <Content style={styles.drawerContent}>
          <ListItem
            last
            style={styles.listItem}
            button
            onPress={() => {
              navigation.navigate('Expenses');
            }}
            iconLeft>
            <Icon name="ios-cash-outline" style={styles.linkIcon} />
            <Text style={styles.linkText}>My Expenses</Text>
          </ListItem>
          <ListItem
            last
            style={styles.listItem}
            button
            onPress={() => {
              navigation.navigate('Categories');
            }}
            iconLeft>
            <Icon name="ios-apps-outline" style={styles.linkIcon} />
            <Text style={styles.linkText}>Categories</Text>
          </ListItem>
          <ListItem
            last
            style={styles.listItem}
            button
            onPress={() => {
              navigation.navigate('Overview');
            }}
            iconLeft>
            <Icon name="ios-stats-outline" style={styles.linkIcon} />
            <Text style={styles.linkText}>Overview</Text>
          </ListItem>
          <ListItem
            last
            style={styles.listItem}
            button
            onPress={() => {
              navigation.navigate('Calendar');
            }}
            iconLeft>
            <Icon name="ios-calendar-outline" style={styles.linkIcon} />
            <Text style={styles.linkText}>Calendar</Text>
          </ListItem>
          <ListItem
            last
            style={styles.listItem}
            button
            onPress={() => {
              navigation.navigate('Search');
            }}
            iconLeft>
            <Icon name="ios-search-outline" style={styles.linkIcon} />
            <Text style={styles.linkText}>Search & Export</Text>
          </ListItem>
          <ListItem
            last
            style={styles.listItem}
            button
            onPress={() => {
              navigation.navigate('Profile');
            }}
            iconLeft>
            <Icon name="ios-person-outline" style={styles.linkIcon} />
            <Text style={styles.linkText}> Profile</Text>
          </ListItem>
          <ListItem
            last
            style={styles.listItem}
            button
            onPress={() => {
              navigation.navigate('Settings');
            }}
            iconLeft>
            <Icon name="ios-options-outline" style={styles.linkIcon} />
            <Text style={styles.linkText}>Settings</Text>
          </ListItem>
        </Content>
        <View style={styles.logoutContainer}>
          <View style={styles.logoutbtn}>
            <Grid>
              <Col>
                <TouchableOpacity
                  onPress={() => {
                    navigation.dispatch(resetAction);
                  }}
                  style={{
                    alignSelf: 'flex-start',
                    backgroundColor: 'transparent',
                  }}>
                  <Text style={{ fontWeight: 'bold', color: '#444' }}>
                    LOG OUT
                  </Text>
                  <Text note>John Doe</Text>
                </TouchableOpacity>
              </Col>
              <Col>
                <TouchableOpacity
                  style={{ alignSelf: 'flex-end' }}
                  onPress={() => {
                    navigation.navigate('Profile');
                  }}>
                  <Thumbnail
                    source={require('../../../assets/Avatars/john.png')}
                    style={styles.profilePic}
                  />
                </TouchableOpacity>
              </Col>
            </Grid>
          </View>
        </View>
      </Container>
    );
  }
}

export default SideBar;
