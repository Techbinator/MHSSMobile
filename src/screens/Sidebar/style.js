const React = require('react-native');

const { Platform } = React;

export default {
  drawerContent: {
    paddingTop: Platform.OS === 'android' ? 20 : 30,
    backgroundColor: '#fff',
  },
  listItem: {
    height: 70,
  },
  linkText: {
    paddingLeft: 15,
    color: '#444',
  },
  linkIcon: {
    color: '#444',
    fontSize: Platform.OS === 'android' ? 25 : 30,
  },
  logoutContainer: {
    padding: 30,
    paddingTop: 0,
    backgroundColor: '#fff',
  },
  logoutbtn: {
    paddingTop: 30,
    flexDirection: 'row',
    borderTopWidth: 1,
    borderTopColor: '#fff',
  },
  profilePic: {
    height: 40,
    width: 40,
    borderRadius: Platform.OS === 'android' ? 40 : 20,
  },
};
