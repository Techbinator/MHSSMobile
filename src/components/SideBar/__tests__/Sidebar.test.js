import React from 'react';
import renderer from 'react-test-renderer';
import { Button, ListItem } from 'native-base';
import mockStore from 'redux-mock-store';
import getElementWithContext from 'react-test-context-provider';

import Sidebar from '../index';

it('renders correctly', () => {
  const tree = renderer.create(<Sidebar />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('navigates to the sub routes', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
  };
  const testRenderer = renderer.create(<Sidebar {...props} />);
  const instance = testRenderer.root;
  const listItems = instance.findAllByType(ListItem);

  listItems.forEach(item => item.props.onPress());

  expect(props.navigation.navigate.mock.calls).toEqual([
    ['Expenses'],
    ['Categories'],
    ['ExpensesCharts'],
    ['ExpenseAgenda'],
    ['Search'],
    ['Profile'],
    ['Settings'],
    ['SignIn'],
  ]);
});
