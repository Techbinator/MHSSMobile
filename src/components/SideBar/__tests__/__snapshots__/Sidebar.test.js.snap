// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "transparent",
            "borderBottomWidth": 0.5,
            "elevation": 0,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
            "top": 0,
          },
          Object {
            "marginTop": 5,
          },
        ]
      }
      transparent={true}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 4,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "opacity": 1,
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                },
                Object {
                  "borderRadius": 20,
                  "height": 40,
                  "width": 40,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        Object {
          "backgroundColor": "#fff",
          "paddingTop": 30,
        },
      ]
    }
    viewIsInsideTabBar={false}
  >
    <View>
      <RCTScrollView
        data={
          Array [
            Object {
              "icon": "ios-cash-outline",
              "route": "Expenses",
              "title": "My Expenses",
            },
            Object {
              "icon": "ios-apps-outline",
              "route": "Categories",
              "title": "Categories",
            },
            Object {
              "icon": "ios-stats-outline",
              "route": "ExpensesCharts",
              "title": "Expenses Analytics",
            },
            Object {
              "icon": "ios-calendar-outline",
              "route": "ExpenseAgenda",
              "title": "Expenses Calendar",
            },
            Object {
              "icon": "ios-search-outline",
              "route": "Search",
              "title": "Search & Export",
            },
            Object {
              "icon": "ios-person-outline",
              "route": "Profile",
              "title": "Profile",
            },
            Object {
              "icon": "ios-options-outline",
              "route": "Settings",
              "title": "Settings",
            },
            Object {
              "icon": "ios-power",
              "route": "SignIn",
              "title": "Log Out",
            },
          ]
        }
        disableVirtualization={false}
        extraData={
          Object {
            "selected": "",
          }
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#c9c9c9",
                      "flexDirection": "row",
                      "marginLeft": 16,
                      "paddingBottom": 13,
                      "paddingRight": 16,
                      "paddingTop": 13,
                    },
                    Object {
                      "borderColor": "rgba(29, 29, 38, 0.1)",
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "paddingLeft": 15,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#444",
                        "fontFamily": "Roboto_light",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  My Expenses
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#c9c9c9",
                      "flexDirection": "row",
                      "marginLeft": 16,
                      "paddingBottom": 13,
                      "paddingRight": 16,
                      "paddingTop": 13,
                    },
                    Object {
                      "borderColor": "rgba(29, 29, 38, 0.1)",
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "paddingLeft": 15,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#444",
                        "fontFamily": "Roboto_light",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Categories
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#c9c9c9",
                      "flexDirection": "row",
                      "marginLeft": 16,
                      "paddingBottom": 13,
                      "paddingRight": 16,
                      "paddingTop": 13,
                    },
                    Object {
                      "borderColor": "rgba(29, 29, 38, 0.1)",
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "paddingLeft": 15,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#444",
                        "fontFamily": "Roboto_light",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Expenses Analytics
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#c9c9c9",
                      "flexDirection": "row",
                      "marginLeft": 16,
                      "paddingBottom": 13,
                      "paddingRight": 16,
                      "paddingTop": 13,
                    },
                    Object {
                      "borderColor": "rgba(29, 29, 38, 0.1)",
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "paddingLeft": 15,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#444",
                        "fontFamily": "Roboto_light",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Expenses Calendar
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#c9c9c9",
                      "flexDirection": "row",
                      "marginLeft": 16,
                      "paddingBottom": 13,
                      "paddingRight": 16,
                      "paddingTop": 13,
                    },
                    Object {
                      "borderColor": "rgba(29, 29, 38, 0.1)",
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "paddingLeft": 15,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#444",
                        "fontFamily": "Roboto_light",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Search & Export
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#c9c9c9",
                      "flexDirection": "row",
                      "marginLeft": 16,
                      "paddingBottom": 13,
                      "paddingRight": 16,
                      "paddingTop": 13,
                    },
                    Object {
                      "borderColor": "rgba(29, 29, 38, 0.1)",
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "paddingLeft": 15,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#444",
                        "fontFamily": "Roboto_light",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Profile
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#c9c9c9",
                      "flexDirection": "row",
                      "marginLeft": 16,
                      "paddingBottom": 13,
                      "paddingRight": 16,
                      "paddingTop": 13,
                    },
                    Object {
                      "borderColor": "rgba(29, 29, 38, 0.1)",
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "paddingLeft": 15,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#444",
                        "fontFamily": "Roboto_light",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Settings
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 0.5,
                      "borderColor": "#c9c9c9",
                      "flexDirection": "row",
                      "marginLeft": 16,
                      "paddingBottom": 13,
                      "paddingRight": 16,
                      "paddingTop": 13,
                    },
                    Object {
                      "borderColor": "rgba(29, 29, 38, 0.1)",
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "paddingLeft": 15,
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "alignSelf": "center",
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#444",
                        "fontFamily": "Roboto_light",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Log Out
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
